### PR TITLE
DevWs for CoolKit Technology Co.,Ltd containing containing eWeLink Button

### PR DESCRIPTION
**Pull request by mingwei0827**

### CODE REVIEW

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 1]
```diff
@@ -415,6 +446,18 @@ private boolean isSomfySituo4() {
...
+}
+
+private boolean isEWeLinkWb01() {
+	isEWeLink() && device.getDataValue("model") == "WB01"
+}
+
+private boolean isEWeLinkSnzb01p() {
+	isEWeLink() && device.getDataValue("model") == "SNZB-01P"
+}
+
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: Couldn't we just get away with `isEWeLink()` and removing model checks, as there's no any device specific configuration and only some exceptions for all manufacturer's devices?

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 2]
```diff
@@ -374,6 +386,25 @@ private Map getButtonEvent(Map descMap) {
...
+			} 
+			else if (descMap.commandInt == 0x01)
+			{
+				buttonState = "double"
+			}
+			else
+			{
+				buttonState = "pushed"
+			}
+		}
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: Please use
```
if (condition) {
	// statement
} else if (anotherCondition) {
	// statement
} else {
	// default statement
}
```
style.

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy (hunk 2 above)
Comment (mingwei0827) [reply devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: Hi, I have modified the code. Please review, thank you.

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 3]
```diff
@@ -374,6 +386,20 @@ private Map getButtonEvent(Map descMap) {
...
+			if (descMap.commandInt == 0x00) {
+				buttonState = "held"
+			} 
+			else if (descMap.commandInt == 0x01) {
+				buttonState = "double"
+			}
+			else {
+				buttonState = "pushed"
+			}
+		}
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy:402]: Please correct `if / else` block styling to the same I've asked for.
Quoted code should look like this
```
	} else if (isEWeLink()) {
		if (descMap.clusterInt == zigbee.ONOFF_CLUSTER) {
			buttonNumber = 1
			if (descMap.commandInt == 0x00) {
				buttonState = "held"
			} else if (descMap.commandInt == 0x01) {
				buttonState = "double"
			} else {
				buttonState = "pushed"
			}
		}
```

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]

**Pull request closed without merging**